examples/LAparser.py

```diff
@@ -405,153 +405,21 @@ def parser(mo):
-        (
-            "Complicated expression",
-            "myscalar = (M3_amatrix * V3_bvector)^Mag + 5*(-xyz[i] + 2.03^2)",
-            "myscalar=(sqrt(vMagnitude2(mvMultiply(amatrix,bvector)))+5*(-xyz[i]+pow(2.03,2)))",
-        ),
-        (
-            "Complicated Multiline",
-            "myscalar = \n(M3_amatrix * V3_bvector)^Mag +\n 5*(xyz + 2.03^2)",
-            "myscalar=(sqrt(vMagnitude2(mvMultiply(amatrix,bvector)))+5*(xyz+pow(2.03,2)))",
-        ),
-    ]
-
-    all_passed = [True]
-
-    def post_test(test, parsed):
-
-        # copy exprStack to evaluate and clear before running next test
-        parsed_stack = exprStack[:]
-        exprStack.clear()
-
-        name, testcase, expected = next(tc for tc in testcases if tc[1] == test)
-
-        this_test_passed = False
-        try:
-            try:
-                result = _evaluateStack(parsed_stack)
-            except TypeError:
-                print(
-                    "Unsupported operation on right side of '%s'.\nCheck for missing or incorrect tags on non-scalar operands."
-                    % input_string,
-                    file=sys.stderr,
-                )
-                raise
-            except UnaryUnsupportedError:
-                print(
-                    "Unary negation is not supported for vectors and matrices: '%s'"
-                    % input_string,
-                    file=sys.stderr,
-                )
-                raise
-
-            # Create final assignment and print it.
-            if debug_flag:
-                print("var=", targetvar)
-            if targetvar != None:
-                try:
-                    result = _assignfunc(targetvar, result)
-                except TypeError:
-                    print(
-                        "Left side tag does not match right side of '%s'"
-                        % input_string,
-                        file=sys.stderr,
-                    )
-                    raise
-                except UnaryUnsupportedError:
-                    print(
-                        "Unary negation is not supported for vectors and matrices: '%s'"
-                        % input_string,
-                        file=sys.stderr,
-                    )
-                    raise
-
-            else:
-                print("Empty left side in '%s'" % input_string, file=sys.stderr)
-                raise TypeError
-
-            parsed["result"] = result
-            parsed["passed"] = this_test_passed = result == expected
-
-        finally:
-            all_passed[0] = all_passed[0] and this_test_passed
-            print("\n" + name)
-
-    equation.runTests((t[1] for t in testcases), postParse=post_test)
-
-    ##TODO: Write testcases with invalid expressions and test that the expected
-    ## exceptions are raised.
-
-    print("Tests completed!")
-    print("PASSED" if all_passed[0] else "FAILED")
-    assert all_passed[0]
-
-
```

Comment:
Some examples are clearly split into parser-definition and test.  When this happens, an example is split into the two parts.